companies/views.py

```diff
@@ -214,6 +214,18 @@ def add_team(request):
     return render(request, "companies/add_team.html", {"form": form})
 
 
+@login_required
+@is_company_owner
+def delete_team(request, name):
+    if request.method == "POST":
+        try:
+            team = Team.objects.get(company=request.user.company, name=name)
```

Comment:
You can use [get_object_or_404](https://docs.djangoproject.com/en/4.1/topics/http/shortcuts/#get-object-or-404) here which will reduce the amount of code required dramatically.


```
team = get_object_or_404(Team, pk=pk, company=request.user.company)
team.delete()
```